precompile/utils.go

```diff
@@ -34,62 +33,37 @@ func deductGas(suppliedGas uint64, requiredGas uint64) (uint64, error) {
 	return suppliedGas - requiredGas, nil
 }
 
-// packOrderedHashesWithSelector checks fullLength of given [input]
-// it excludes first member since it should be the function selector
-// then checks if the given [fullLength] is a multiple of member count * common.HashLength
-func packOrderedHashesWithSelector(input [][]byte, fullLength int) ([]byte, error) {
-	hashLen := fullLength - selectorLen
-	realLen := (len(input) - 1) * common.HashLength
-	if hashLen != realLen {
-		return nil, fmt.Errorf("expected %d, got %d length", hashLen, realLen)
-	}
-
-	// check function selector
-	if selectorLen != len(input[0]) {
-		return nil, fmt.Errorf("first element of the input must be a function selector with length %d", selectorLen)
-	}
-	// first handle selector
-	buf := make([]byte, fullLength)
-	copy(buf[:selectorLen], input[0])
-
-	// handle bytesHashes
-	bytesHashes, err := packOrderedHashes(input[1:], hashLen)
-	if err != nil {
-		return nil, err
-	}
-	copy(buf[selectorLen:], bytesHashes)
-
-	return buf, nil
+// packOrderedHashesWithSelector packs the function selector and ordered list of hashes into [dst]
+// byte slice.
+// assumes that [dst] has sufficient room for [functionSelector] and [hashes].
+func packOrderedHashesWithSelector(dst []byte, functionSelector []byte, hashes []common.Hash) {
```

Comment:
I was just thinking it's an extension to `packOrderedHashes(dst []byte, hashes []common.Hash)` , so it would take functionSelector as the extra param. but that's really a nit